docs/guide/mocking-requests.md

```diff
@@ -0,0 +1,57 @@
+# Mocking Requests
+
+We write tests in order to be able to ship with confidence. The more underlying behavior we mock, the less confidence we get since the system is less closer to real usage.
+
+Mocking network requests is one of those tricky aspects of running tests in a non-browser environment. Web API's are not available so now we need to mock or install something that mocks this behavior.
+
+There are various tools for this job, but which one do you choose?
+
+We recommend [Mock Service Worker](https://mswjs.io/) to mock network requests. It handles both `REST` and `GraphQL` type of network requests and is framework agnostic.
+
+Mock Service Worker (or MSW in short) uses [Service Worker API](https://developers.google.com/web/fundamentals/primers/service-workers) to intercept requests. It has a variety of use cases but also offers functionality specifically for `node`. It won't magically make `service workers API` available for non-browser environments, but instead relies on [node-request-interceptor](https://mswjs.io/docs/api/setup-server#operation).
```

Comment:
```suggestion
Mock Service Worker (MSW) works by intercepting the requests your tests make, allowing you to use it without changing any of your application code. In-browser, this uses the [Service Worker API](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API). In Node.js, and for Vitest, it uses [node-request-interceptor](https://mswjs.io/docs/api/setup-server#operation). To learn more about MSW, read their [introduction](https://mswjs.io/docs/)
```

Reply:
I re-read the MSW docs and looked at the code samples; I think the description of it using the Service Worker API isn't quite right; I think this is probably a more accurate description that both shows readers where the name came from and explains how it works for Node